submitr/rclone/rclone_amazon.py

```diff
@@ -0,0 +1,199 @@
+from __future__ import annotations
+from boto3 import client as BotoClient
+import configparser
+from typing import Optional, Tuple, Union
+from dcicutils.file_utils import normalize_path
+from dcicutils.misc_utils import create_dict, normalize_string
+from submitr.rclone.rclone_config import RCloneConfig, RCloneCredentials
+from submitr.rclone.rclone_utils import cloud_path
+
+
+class RCloneAmazon(RCloneConfig):
```

Comment:
Yeah I originally had it named that, and changed because it's not *really* just a config - it has calls to do things - e.g. to get the file size and checksum - So I feel the abstract base class is actually misnamed and it should maybe be RCloneTarget  and these should be either like RCloneAmazon or RCloneTargetAmazon. Unless objections I think I'll rename RCloneConfig to RCloneTarget - I think then RCloneAmazon/Google will make more sense.